### DISCUSSION TITLE
WSL(Windows Linux) has a bug that cannot install docker. Do NOT use WSL to run CML nodes

### DISCUSSION THREAD

**Discussion by kevingzhang** [tearust/teaproject#103] [Bug report]
We just received a report that WSL (A linux version run above Windows) cannot install docker. The reason is that WSL 1 is not a real Linux. It doesn't support domain socket which Docker and TEA mining depend on. You will see an error message
` Docker: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?`
There is no solution from our side so far. Someone says WSL2 fixed this issue but we cannot verify since we do not have a Windows machine.

I would suggest avoid using any Windows modified Linux. Please either install a Ubuntu on your hardware, or a real Ubuntu virtual machine. Someone is trying Multipass on Windows. We have not tested it either. the result ti unknown at this moment.

**Comment by kevingzhang** [comment]
@geopor Please update our wiki to make sure they do not try Windows based Linux

**Reply by geopor** [reply]
Added explanation with link to it from original hardware mining guide: https://github.com/tearust/teaproject/wiki/Mining-With-Your-Own-Hardware---Virtual-Machines#mac--windows-compatibility-for-mining-machine-script
